src/Schemer/Validators/Inputs/BasicInput.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ * Schemer
+ * @author Roman Pistek
+ */
+
+namespace Schemer\Validators\Inputs;
+
+use Schemer\Validators\Input;
+use Schemer\Support\Strings;
+use Schemer\Exceptions\InvalidUserInputException;
+use Nette\Utils\Callback;
+use Nette\InvalidArgumentException;
+
+
+/**
+ * Basic user input validator
+ *
+ * @author Roman Pistek
+ */
+class BasicInput implements Input
```

Comment:
```suggestion
abstract class BasicInput implements Input
```
a ostatní nastavit na final?

Reply:
Done